### PR TITLE
Tensor contraction pr

### PR DESCRIPTION
**Pull request by jichangjichang**
Fixes for MatrixInstruction
- use correct StridesC for multi-dim tensor 
- support shiftPtr for any glvw

### CODE REVIEW

**Comment by jichangjichang** [comment]
rocblas tests all pass
[==========] 1439368 tests from 469 test cases ran. (37617392 ms total)
[  PASSED  ] 1439368 tests.

**Comment by jichangjichang** [comment]
> pyflakes isn't happy...
@zaliu 
how you run the pyflakes ?
I ran it but the only error is not related to this PR.
 
python3 -m pyflakes KernelWriterAssembly.py
KernelWriterAssembly.py:8274: local variable 'startStride' is assigned to but never used

File: Tensile/KernelWriterAssembly.py [hunk 1]
```diff
@@ -9231,31 +9234,31 @@ def emitAddressSetupCode(self, kernel, ss, tmpVgpr01, tmpS01, edge, beta, atomic
...
           sTmp2 = tmpS01+2
-          kStr += inst("v_cmp_eq_u32", sgpr(sTmp1,2), vgpr(self.coord1Vgpr), \
-                        sgpr("SizesFree+1"), "if coord1 is edge+1")
-          kStr += inst("v_and_b32", vgpr(vTmp1), vgpr(self.coord1Vgpr), vw-1, "mod VW")
-          kStr += inst("v_cmp_eq_u32", sgpr(sTmp2,2), vgpr(vTmp1), vw-1, "if coord1-1 is odd edge")
-          kStr += inst("s_and_b64", sgpr(sTmp1,2), sgpr(sTmp1,2), sgpr(sTmp2,2), "if meet both conditions")
-          kStr += inst("s_lshr_b64", sgpr(sTmp2,2), sgpr(sTmp1,2), vw-1, "if threadx meet conditions, set threadx-1 bit in another sgprs")
-          kStr += inst("v_sub_u32", vgpr(vTmp1), vgpr(self.coord1Vgpr), 1, "tmp = coord1 - 1")
-          kStr += inst("v_add_u32", vgpr(vTmp2), vgpr(self.coord1Vgpr), 1, "tmp = coord1 + 1")
+          # calculat condition
```
Comment (bensander) [inline Tensile/KernelWriterAssembly.py]: calculate typo

File: Tensile/KernelWriterAssembly.py [hunk 2]
```diff
@@ -9231,31 +9234,31 @@ def emitAddressSetupCode(self, kernel, ss, tmpVgpr01, tmpS01, edge, beta, atomic
...
+          kStr += inst("v_bfi_b32", vgpr(vTmp1), vw-1, 0, vgpr(self.coord1Vgpr), "coord1 & ~(vw-1)")
+          kStr += inst("v_bfi_b32", vgpr(vTmp2), vw-1, 0, sgpr("SizesFree+1"), "sizeFree1 & ~(vw-1)")
+          kStr += inst("v_cmp_eq_u32", sgpr(sTmp1,2), vgpr(vTmp1), vgpr(vTmp2), "if coord1 is in edge glvw")
+          kStr += inst("v_and_b32", vgpr(vTmp2), sgpr("SizesFree+1"), vw-1, "sizeFree1 mod VW")
+          kStr += inst("v_cmp_gt_u32", sgpr(sTmp2,2), vgpr(vTmp2), 0, "this problem is not multiple size of glvw")
+          kStr += inst("s_and_b64", sgpr(sTmp1,2), sgpr(sTmp1,2), sgpr(sTmp2,2), "AND both conditions")
+          # calculat new coord
```
Comment (bensander) [inline Tensile/KernelWriterAssembly.py]: calculate

**Pull request merged**